tests/pii_replacer/replacement/test_scope.py

```
@@ -0,0 +1,482 @@
+    assert dob_spec.entity_type == PiiEntity.date_of_birth
+
+    replacer = TabularPiiReplacer(config, data_config=data_config)
+    replacer.transform_df(df)
+    assert replacer.result is not None
+    out = replacer.result.transformed_df
+
+    # patient_id fully transformed, consistent within each group, and globally
+    # unique (identifiers get a hard cross-group uniqueness guarantee).
+    assert (out["patient_id"] != df["patient_id"]).all()
+    orig_to_new: dict[str, str] = {}
+    for o, n in zip(df["patient_id"], out["patient_id"]):
```

Ruff (0.16.1)

[warning] `zip()` without an explicit `strict=` parameter

Add explicit value for parameter `strict=`

(B905)